### PR TITLE
[RSS] Get ready for content collections

### PR DESCRIPTION
**Pull request by bholmesdev**
## Changes

- Expose `rssSchema` to type-check content collections for RSS feeds
- Deprecate internal `import.meta.glob` handling and move to a `pagesGlobToRssItems()` helper. This has a few benefits:
  - Simplifies our `items` array to accept a single type. Better for JS docs and docs.astro.build.
  - Makes the "only glob `src/pages/` entries" requirement more explicit with the function name.
  - Makes handling globs and handling content collections feel equally supported. When `import.meta.glob` handling was implicit, this felt like the more encouraged solution.
- Refactor internals to use Zod all the way down. This makes error handling on bad config much clearer!

### README changes

- Change glob example to a content collections example
- Add `rssSchema` and `pagesGlobToRssItems()` reference
- Chore: change `import.meta.env.SITE` recommendation to new `context.site`

## Testing

- Update testing for new APIs
- Migrate glob testing to separate `pagesGlobToRssItems()` tests

## Docs

- update README
- update docs and tutorial https://github.com/withastro/docs/pull/2401

### CODE REVIEW

**Comment by bholmesdev** [comment]
Updating to keep `import.meta.glob()` handling around as a deprecated API, and dropping to a non-breaking minor!

File: .changeset/fluffy-cups-travel.md
Comment (natemoo-re) [inline .changeset/fluffy-cups-travel.md:5]: I'd suggest putting these points last and moving the migration guide first. Coming from someone who is migrating from the previous major to this major, I'd want to know the relevant breaking changes that I need to make first. I might not know what "content collections" is.

File: packages/astro-rss/src/index.ts
Comment (natemoo-re) [inline packages/astro-rss/src/index.ts:60]: Typo?
```suggestion
	draft: z.boolean().default(false),
```

**Review by natemoo-re — COMMENTED** [review]
Seems like a great refactor! Had a few comments around messaging and maybe found a typo?

File: packages/astro-rss/src/index.ts
Comment (bholmesdev) [reply packages/astro-rss/src/index.ts:76]: Actually not! This mirrors the naming of `markdown.drafts`, aka "include drafts yes or no." I fought to rename to `includeDrafts` and lost 😅

File: .changeset/fluffy-cups-travel.md
Comment (bholmesdev) [reply .changeset/fluffy-cups-travel.md]: Understood. Redrafted and bumped down to a minor! This is now deprecated, not removed 👍

**Review by sarah11918 — COMMENTED** [review]
Quick Docs README review:

I've reviewed the accompanying PR to docs, which I know contains some of the same examples and sections. I'll let you look over those comments, and decide which ones are applicable here. It's OK if this README is more technical than the docs RSS page, which should be more oriented to practical usage. 

But, I want to give you a chance to look over that review and 
a) see if there's any suggestion there that you'd choose to carry into here. 
b) decide whether there's anything you're happy to leave exclusively to docs, and cut from the README. (We favour as little duplication as possible, and keeping the READMES to a minimum, where it makes sense. This document won't often have to stand on its own since readers are far more likely to encounter the docs page, and search only works in docs proper, etc.)

Then, I'll take another last look over this, but I don't see anything that's going to hold this up!

File: packages/astro-rss/README.md
Comment (sarah11918) [inline packages/astro-rss/README.md:184]: WE suggest using a package... (or, whatever fix you made in the Docs PR)

File: packages/astro-rss/README.md
Comment (sarah11918) [inline packages/astro-rss/README.md:184]:
```suggestion
**Note:** Whenever you're using HTML content in XML, we suggest using a package like [`sanitize-html`](https://www.npmjs.com/package/sanitize-html) in order to make sure that your content is properly sanitized, escaped, and encoded.
```

**Review by sarah11918 — APPROVED** [review]
Just re-add that missing "we" and good to go from Docs, @bholmesdev!

**Pull request merged**
